PhysicsTools/PatAlgos/plugins/TauJetCorrFactorsProducer.cc

```diff
@@ -162,37 +156,31 @@ void TauJetCorrFactorsProducer::produce(edm::Event& evt, const edm::EventSetup&
           << "This makes no sense, either you should correct this or drop the module from \n"
           << "the sequence.";
 
-    std::string payload = defaultPayload_;
-    if (dynamic_cast<const reco::PFTau*>(&(*tauJet))) {
-      const reco::PFTau* pfTauJet = dynamic_cast<const reco::PFTau*>(&(*tauJet));
-      for (std::vector<payloadMappingType>::const_iterator payloadMapping = payloadMappings_.begin();
-           payloadMapping != payloadMappings_.end();
-           ++payloadMapping) {
-        for (vint::const_iterator decayMode = payloadMapping->decayModes_.begin();
-             decayMode != payloadMapping->decayModes_.end();
-             ++decayMode) {
-          if (pfTauJet->decayMode() == (*decayMode))
-            payload = payloadMapping->payload_;
+    auto const* payloadToken = &defaultPayloadToken_;
+    if (auto pfTauJet = dynamic_cast<const reco::PFTau*>(&tauJet)) {
+      for (auto const& payloadMapping : payloadMappings_) {
+        for (auto const& decayMode : payloadMapping.decayModes_) {
+          if (pfTauJet->decayMode() == decayMode)
+            payloadToken = &payloadMapping.payloadToken_;
         }
       }
     }
 
     // retrieve JEC parameters from the DB and build a new corrector,
     // in case it does not exist already for current payload
-    if (correctorMapping.find(payload) == correctorMapping.end()) {
-      edm::ESHandle<JetCorrectorParametersCollection> jecParameters;
-      es.get<JetCorrectionsRecord>().get(payload, jecParameters);
-
-      correctorMapping[payload] = std::make_shared<FactorizedJetCorrector>(params(*jecParameters, levels_));
+    const int payloadTokenValue = payloadToken->index().value();
```

Comment:
Okay right, but this commit is outdated now because the esConsumes migration of the `TauJetCorrFactorsProducer` was done in https://github.com/cms-sw/cmssw/pull/34997.